daemon/logger/fluentd/fluentd.go

```diff
@@ -52,7 +52,6 @@ const (
 
 	addressKey                = "fluentd-address"
 	asyncKey                  = "fluentd-async"
-	asyncConnectKey           = "fluentd-async-connect" // deprecated option (use fluent-async instead)
```

Comment:
We probably need a migration step for this on container restore. If we remove this option here, I _think_ existing containers that have the option set will fail to start when starting the daemon after updating.

So in the "restore" part of the daemon (where we restore containers from their config on-disk), we need to patch existing containers and remove the `fluentd-async-connect`, and set the `fluentd-async` option instead (then save the container config).

We had a similar case for the log entries driver (but also needs to be fixed before we can merge that one);

- https://github.com/moby/moby/pull/44442